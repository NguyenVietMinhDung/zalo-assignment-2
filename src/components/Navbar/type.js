export type Item = {
  id: string,
  name: string,
};

export type Props = {
  items: Array<Item>,
};
