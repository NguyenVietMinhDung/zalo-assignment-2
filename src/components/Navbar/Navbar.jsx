// @flow
import React from 'react';
import Scrollspy from 'react-scrollspy';
import './Navbar.scss';
import { Props } from './type';
import CategoryLink from './CategoryLink';

const Navbar = (props: Props) => {
  const { items } = props;
  return (
    <div className="navbar">
      <Scrollspy items={items.map(item => item.id)} className="navbar__menu" currentClassName="is-current">
        {items.map(item => (
          <CategoryLink key={item.id} {...item} />
        ))}
      </Scrollspy>
      <div className="navbar__search">
        <i className="lni lni-32 lni-search-alt" />
      </div>
    </div>
  );
};

export default Navbar;
