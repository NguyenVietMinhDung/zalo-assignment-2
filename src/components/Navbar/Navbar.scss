@import '../../styles/variables';

.navbar {
	position: sticky;

	z-index: 1;
	top: 0;

	display: flex;

	background-color: $light-red;

	&__menu {
		display: flex;
		overflow: auto;
		flex: 1;

		height: 50px;

		margin: 0;
		padding: 0 10px;

		white-space: nowrap;

		.is-current {
			a {
				color: white;
			}

			&::after {
				position: absolute;
				bottom: 0;

				width: 100%;
				height: 4px;

				content: '';

				background-color: orange;
			}
		}
	}

	&__search {
		display: flex;

		padding: 0 10px;

		color: $light-pink;
		border-left: 1px solid $dark-red;
		align-items: center;
		justify-content: center;

		&:hover {
			cursor: pointer;

			color: white;
		}
	}
}
