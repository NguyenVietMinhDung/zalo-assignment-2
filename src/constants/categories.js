export default [
  {
    id: 'other',
    name: 'Khác',
    items: [
      {
        id: 0,
        name: {
          vi: 'Phin Inox',
          en: 'Phin Inox',
        },
        price: 80000,
        img: '270_crop_PHININOX.png',
      },
      {
        id: 1,
        name: {
          vi: 'Ly Sứ Mosaic',
          en: 'Mosaic Mug',
        },
        price: 29000,
        img: '270_crop_LYSU.png',
      },
    ],
  },
  {
    id: 'bread',
    name: 'Bánh Mì',
    items: [
      {
        id: 0,
        name: {
          vi: 'Xíu Mại',
          en: 'Pork Meatball',
        },
        price: 19000,
        img: '270_crop_BMXIUMAI.png',
      },
      {
        id: 1,
        name: {
          vi: 'Chả Lụa Xá Xíu',
          en: 'Roast Pork & Pork Sausage',
        },
        price: 19000,
        img: '270_crop_BMCHALUAXAXIU.png',
      },
      {
        id: 2,
        name: {
          vi: 'Gà Xé Nước Tương',
          en: 'Roast Chicken',
        },
        price: 19000,
        img: '270_crop_BMGAXE.png',
      },
      {
        id: 3,
        name: {
          vi: 'Thịt Nướng',
          en: 'Grilled Pork',
        },
        price: 19000,
        img: '270_crop_BMTHITNUONG.png',
      },
    ],
  },
  {
    id: 'cake',
    name: 'Bánh Ngọt',
    items: [
      {
        id: 0,
        name: {
          vi: 'Bánh Mousse Cacao',
          en: 'Cocoa Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSECACAO.png',
      },
      {
        id: 1,
        name: {
          vi: 'Bánh Mousse Đào',
          en: 'Peach Mousse Cake',
        },
        price: 29000,
        img: '270_crop_MOUSSEDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Bánh Sô-cô-la Highlands',
          en: 'Highlands Chocolate Bomb',
        },
        price: 29000,
        img: '270_crop_SOCOLAHL.png',
      },
    ],
  },
  {
    id: 'coffee',
    name: 'Cà Phê Đóng Gói',
    items: [
      {
        id: 0,
        name: {
          vi: 'Latte',
          en: 'Latte',
        },
        price: 60000,
        img: '270_crop_LATTE.png',
      },
      {
        id: 1,
        name: {
          vi: 'Cappuccino',
          en: 'Cappuccino',
        },
        price: 60000,
        img: '270_crop_CAPPUCINO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Americano',
          en: 'Americano',
        },
        price: 60000,
        img: '270_crop_AMERICANO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Espresso',
          en: 'Espresso',
        },
        price: 60000,
        img: '270_crop_ESPRESSO.png',
      },
      {
        id: 4,
        name: {
          vi: 'Caramel Macchiato',
          en: 'Caramel Macchiato',
        },
        price: 60000,
        img: '270_crop_CARAMEL-MACCHIATO.png',
      },
    ],
  },
  {
    id: 'Tea',
    name: 'Trà',
    items: [
      {
        id: 0,
        name: {
          vi: 'Trà Thạch Vãi',
          en: 'Lychee Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHVAI_1.png',
      },
      {
        id: 1,
        name: {
          vi: 'Trà Thạch Đào',
          en: 'Peach Jelly Tea',
        },
        price: 39000,
        img: '270_crop_TRATHACHDAO.png',
      },
      {
        id: 2,
        name: {
          vi: 'Trà Thanh Đào',
          en: 'Peach Lemongrass Tea',
        },
        price: 39000,
        img: '270_crop_TRATHANHDAO.png',
      },
      {
        id: 3,
        name: {
          vi: 'Trà Sen Vàng',
          en: 'Golden Lotus Tea',
        },
        price: 39000,
        img: '270_crop_TRASENVANG.png',
      },
    ],
  },
];
