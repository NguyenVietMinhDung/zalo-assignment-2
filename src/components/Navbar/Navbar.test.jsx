import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';
import categories from '../../constants/categories';

const navbarItems = categories.map(category => ({ id: category.id, name: category.name }));

const defaultProps = {
  items: navbarItems,
};

describe('<Navbar />', () => {
  test('renders correctly', () => {
    const component = shallow(<Navbar {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
