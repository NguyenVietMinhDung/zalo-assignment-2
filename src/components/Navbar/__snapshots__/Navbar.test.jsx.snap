// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<div
  className="navbar"
>
  <Scrollspy
    className="navbar__menu"
    componentTag="ul"
    currentClassName="is-current"
    items={
      Array [
        "other",
        "bread",
        "cake",
        "coffee",
        "Tea",
      ]
    }
    offset={0}
    onUpdate={[Function]}
    style={Object {}}
  >
    <CategoryLink
      id="other"
      key="other"
      name="Khác"
    />
    <CategoryLink
      id="bread"
      key="bread"
      name="Bánh Mì"
    />
    <CategoryLink
      id="cake"
      key="cake"
      name="Bánh Ngọt"
    />
    <CategoryLink
      id="coffee"
      key="coffee"
      name="Cà Phê Đóng Gói"
    />
    <CategoryLink
      id="Tea"
      key="Tea"
      name="Trà"
    />
  </Scrollspy>
  <div
    className="navbar__search"
  >
    <i
      className="lni lni-32 lni-search-alt"
    />
  </div>
</div>
`;
